src/btree/bt_delete.c

```diff
@@ -131,7 +131,7 @@ __wt_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
     /*
      * Fast check to see if it's worth locking, then atomically switch the page's state to lock it.
      */
-    previous_state = ref->state;
+    previous_state = __wt_ref_get_state(ref);
```

Comment:
This switch case is uneeded complexity for what should be an `if (__wt_ref_get_state(ref) != WT_REF_DISK) return (0);`

Reply:
Updated in [d2c93e2](https://github.com/wiredtiger/wiredtiger/pull/10345/commits/d2c93e207197226e60cacfaf2108061f1058f105). I've kept `previous_state = __wt_ref_get_state(ref);` since we also use `previous_state` in the following cas.